autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/persister/CrendentialsDedupStrategy.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.sync.persister
+
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.sync.CredentialsSync
+import com.duckduckgo.autofill.sync.CredentialsSyncMapper
+import com.duckduckgo.autofill.sync.CrendentialsSyncEntries
+import com.duckduckgo.autofill.sync.isDeleted
+import com.duckduckgo.sync.api.engine.SyncMergeResult
+import com.duckduckgo.sync.api.engine.SyncMergeResult.Error
+import com.duckduckgo.sync.api.engine.SyncMergeResult.Success
+import kotlinx.coroutines.runBlocking
+import timber.log.Timber
+
+class CrendentialsDedupStrategy constructor(
+    private val credentialsSync: CredentialsSync,
+    private val credentialsSyncMapper: CredentialsSyncMapper,
+    private val dispatchers: DispatcherProvider,
+) : CredentialsMergeStrategy {
+    override fun processEntries(credentials: CrendentialsSyncEntries): SyncMergeResult<Boolean> {
```

Comment:
Same comment as on the other merge strategy; having everything in one large function is hurting readability of the code IMO